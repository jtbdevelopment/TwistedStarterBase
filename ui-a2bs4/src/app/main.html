<div class="main-container">
  <main class="main">
  </main>
  <fountain-footer></fountain-footer>
</div>