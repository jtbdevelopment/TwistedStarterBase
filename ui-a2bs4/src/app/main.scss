@import '~bootstrap/scss/bootstrap.scss';

.jumbotron {
  .small-screen-warning {
    font-size: 48pt;
    @include media-breakpoint-up(sm) {
      display: none;
    }
  }
}
