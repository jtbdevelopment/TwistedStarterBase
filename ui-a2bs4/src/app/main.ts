import {Component} from '@angular/core';

@Component({
  selector: 'fountain-app',
  templateUrl: './main.html',
  styleUrls: ['./main.scss']
})
export class MainComponent {}
