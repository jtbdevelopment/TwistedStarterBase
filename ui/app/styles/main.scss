$icon-font-path: "../bower_components/bootstrap-sass-official/assets/fonts/bootstrap/";
// bower:scss
@import "bootstrap-sass-official/assets/stylesheets/_bootstrap.scss";
// endbower
@import "colors";
@import "game-bar.scss";
@import "create.scss";
@import "player-list";
@import "admin";
@import "profile";
@import "play";
@import "customfont";

body {
  background-color: $peachOrange;
  padding-top: 10px;
  padding-bottom: 45px;

  .navbar {
    padding-left: 5px;
    padding-right: 5px;
    background-color: $mediumSpringBud;
    border-width: 0;
  }

  .modal-dialog {
    .modal-content {
      background-color: $mediumChampagne;
    }
    .modal-header {
      background-color: $peachOrange;
      border-width: 0;
    }
    .modal-body {
    }
    .modal-footer {
      border-width: 0;
    }

    .btn-danger {
      background-color: $puceRed;
      color: white;
      border-color: $mediumChampagne;
    }
    .btn-danger:hover {
      border-color: $cambridgeBlue;
    }

    .cancel-button {
      background-color: $mediumSpringBud;
      color: white;
      border-color: $mediumChampagne;
    }
    .cancel-button:hover {
      border-color: $cambridgeBlue;
    }
  }
}

.navbar-default .navbar-nav > li > a, .navbar-default .navbar-brand {
  color: $puceRed;
  font-weight: bolder;
}

.pull-left {
  padding-left: 5px;
}

.btn-quit-game {
  background-color: $puceRed;
}

.btn-quit-game:hover {
  background-color: $puceRed;
  color: $cambridgeBlue;
}

.btn-end-rematch-game {
  background-color: $puceRed;
  color: white;
}

.btn-end-rematch-game:hover {
  background-color: $puceRed;
  color: $cambridgeBlue;
}

.btn-rematch-game {
  background-color: $mediumSpringBud;
  color: white;
}

.btn-rematch-game:hover {
  background-color: $mediumSpringBud;
  color: $cambridgeBlue;
}
