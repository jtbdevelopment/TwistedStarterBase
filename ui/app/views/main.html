<div class="welcome-landing">
    <div class="jumbotron">
        <!-- TODO - after adding tracking - use tracking to decide if to say welcome back -->
        <h1>Welcome</h1>
        <p>Click one of your existing games to the left or</p>
        <p>
            <a ng-href="#/create" class="btn btn-lg btn-primary create-game-button">
                <span class="glyphicon glyphicon-plus"></span>
                Start a new game!
            </a>
        </p>
    </div>
</div>