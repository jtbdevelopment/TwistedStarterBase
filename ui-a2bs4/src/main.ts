import { enableProdMode } from '@angular/core';
import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';

import { AppModule } from './app/app.module';
import { environment } from './environments/environment';

import './index.scss';
import 'bootstrap/dist/css/bootstrap.css';
import 'font-awesome/css/font-awesome.css';
import 'primeng/resources/themes/bootstrap/theme.css';
import 'primeng/resources/primeng.css';

if (environment.production) {
  enableProdMode();
}

platformBrowserDynamic().bootstrapModule(AppModule)
  .catch(err => console.log(err));
