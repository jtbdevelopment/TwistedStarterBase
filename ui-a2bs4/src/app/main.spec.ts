import {MainComponent} from './main';
import {TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';

describe('Main Component', () => {
    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [RouterTestingModule],
            declarations: [
                MainComponent
            ]
        });
        TestBed.compileComponents();
    });

    it('should render the header, title, techs and footer', () => {
        const fixture = TestBed.createComponent(MainComponent);
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });
});
