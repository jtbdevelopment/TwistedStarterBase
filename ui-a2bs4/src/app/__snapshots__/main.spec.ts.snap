// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component should render the header, title, techs and footer 1`] = `
<app-main>
  <div
    class="welcome-landing"
  >
    
    
    <div
      class="jumbotron"
    >
      
      
      
      
      <h1>
        Welcome
      </h1>
      
      
      <p>
        Click one of your existing games to the left or
      </p>
      
      
      <p>
        
        
        <a
          class="btn btn-lg btn-primary create-game-button"
          href="/create"
          ng-reflect-router-link="/create"
        >
          
          
          <span
            class="fa fa-plus"
          />
          
          Start a new game!
        
        </a>
        
      
      </p>
      
      
      
      
      <p
        class="small-screen-warning"
      >
        
        Best experience on tablet or desktop.
      
      </p>
      
    
    </div>
    
  
  </div>
</app-main>
`;
